pkg/chunked/storage_linux.go

```diff
@@ -1665,11 +1666,11 @@ func (c *chunkedDiffer) ApplyDiff(dest string, options *archive.TarOptions, diff
 		c.tarSplit = tarSplit
 
 		// since we retrieved the whole file and it was validated, use the diffID instead of the TOC digest.
-		c.contentDigest = diffID
+		c.contentDigest = ""
```

Comment:
AFAICS `contentDigest` is now only ever a TOC digest. So
- Rename it to  something like `tocDigest`, to be precise
- Update the comment documenting the field to not incorrectly talk about DiffIDs
- Remove the comment just above, AFACIS it has no relationship to the code
- Remove this line entirely? For the `…convert…` case, the field is never set. Or maybe set it explicitly to `""` already in the `makeConvertFromRawDiffer` constructor, to be explicit.

Reply:
thanks, I've amended the suggested changes.

I've added a new variable `tocDigest`.  Hopefully, it is clearer than having to change the state of `chunkedDiffer` when both write and read the value within the same function.